### PR TITLE
Add Tensorboard integration

### PR DESCRIPTION
**Pull request by alexpolozov**

### CODE REVIEW

File: chem_tensorflow.py [hunk 1]
```diff
@@ -86,13 +88,18 @@ def __init__(self, args):
...
             # Restore/initialize variables:
             restore_file = args.get('--restore')
             if restore_file is not None:
-                self.restore_model(restore_file)
+                self.train_step_id, self.valid_step_id = self.restore_progress(restore_file)
             else:
                 self.initialize_model()
+                self.train_step_id = 0
+                self.valid_step_id = 0
+            self.train_writer = tf.summary.FileWriter(os.path.join(tb_log_dir, 'train'), graph=self.graph)
```
Comment (mmjb) [inline chem_tensorflow.py:101]: Can we make the use of TensorBoard optional?

**Comment by bzz** [comment]
Sorry for bothering but @mmjb, @mallamanis, do you guys think there is anything missing in order to merge this?

File: chem_tensorflow.py [hunk 2]
```diff
@@ -277,36 +297,39 @@ def train(self):
...
 
     def initialize_model(self) -> None:
         init_op = tf.group(tf.global_variables_initializer(),
                            tf.local_variables_initializer())
         self.sess.run(init_op)
 
-    def restore_model(self, path: str) -> None:
-        print("Restoring weights from file %s." % path)
-        with open(path, 'rb') as in_file:
+    def restore_progress(self, model_path: str) -> (int, int):
```
Comment (mallamanis) [inline chem_tensorflow.py:330]: Please use annotation `Tuple[int, int]` (by importing `Tuple` from `typings`): https://docs.python.org/3/library/typing.html#typing.Tuple

```suggestion
    def restore_progress(self, model_path: str) -> Tuple[int, int]:
```
(and add the import)

**Review by mallamanis — CHANGES_REQUESTED** [review]
Apologies for the delay, somehow this went unnoticed. One minor change and I can merge.

**Pull request merged**